apps/portal/src/content/docs/components/alert-dialog.mdx

```diff
@@ -0,0 +1,341 @@
+
+<DocsPage.ComponentExample
+  client:only
+  code={`<AlertDialog.Root>
+    <AlertDialog.Trigger asChild>
+      <Button>Open Alert Dialog</Button>
+    </AlertDialog.Trigger>
+    <AlertDialog.Content>
+      <AlertDialog.Header>
+        <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
+      </AlertDialog.Header>
+      <AlertDialog.Description variant="secondary">
+        <p>Alert dialog content goes here. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
+      </AlertDialog.Description>
+      <AlertDialog.Footer>
+        <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
+        <AlertDialog.Action>Confirm</AlertDialog.Action>
+      </AlertDialog.Footer>
+    </AlertDialog.Content>
+  </AlertDialog.Root>`}
+/>
+
+## Usage
+
+```typescript jsx
+import { AlertDialog } from '@harnessio/ui/components'
+import { Button } from '@harnessio/ui/components'
+
+return (
+  <AlertDialog.Root>
+    <AlertDialog.Trigger asChild>
+      <Button>Open Alert Dialog</Button>
+    </AlertDialog.Trigger>
+    <AlertDialog.Content>
+      <AlertDialog.Header>
+        <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
+      </AlertDialog.Header>
+      <AlertDialog.Description variant="secondary">
+        This is the alert dialog description.
+        <p>Alert dialog content goes here.</p>
+      </AlertDialog.Description>
+      <AlertDialog.Footer>
+        <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
+        <AlertDialog.Action>Confirm</AlertDialog.Action>
+      </AlertDialog.Footer>
+    </AlertDialog.Content>
+  </AlertDialog.Root>
+)
+```
+
+## Anatomy
+
+All parts of the AlertDialog component can be imported and composed as required.
+
+```typescript jsx
+<AlertDialog.Root>
+  <AlertDialog.Trigger>
+    <Button>Open Alert Dialog</Button>
+  </AlertDialog.Trigger>
+  <AlertDialog.Content>
+    <AlertDialog.Header>
+      <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
+    </AlertDialog.Header>
+    <AlertDialog.Description variant="secondary">
+      This is the alert dialog description.
+      <p>Alert dialog content goes here.</p>
+    </AlertDialog.Description>
+    <AlertDialog.Footer>
+      <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
+      <AlertDialog.Action>Confirm</AlertDialog.Action>
+    </AlertDialog.Footer>
+  </AlertDialog.Content>
+</AlertDialog.Root>
+```
+
+## API Reference
+
+### Root
+
+The `Root` component serves as the main container for all alert dialog elements.
+
+<DocsPage.PropsTable
+  props={[
+    {
+      name: "children",
+      description:
+        "You can pass in your `AlertDialog.Trigger` and `AlertDialog.Content` as children",
+      required: true,
+      value: "ReactNode",
+    },
+  ]}
+/>
+
+### Trigger
+
+The `Trigger` component is used to open the alert dialog.
+
+```typescript jsx
+<AlertDialog.Trigger>
+  <Button>Open Alert Dialog</Button>
+</AlertDialog.Trigger>
```

Comment:
This will render a `button` inside a `button`.
```suggestion
<AlertDialog.Trigger asChild>
  <Button>Open Alert Dialog</Button>
</AlertDialog.Trigger>
```

Or
```suggestion
<AlertDialog.Trigger>
  Open Alert Dialog
</AlertDialog.Trigger>
```